### PR TITLE
 A new middleware function has been added to control the flow of error…(resolving issues #178)

### PR DESCRIPTION
**Pull request by somya339**
…rs and from now onwards the error throwing should follow the decorum mentioned in the documentation

**What kind of change does this PR introduce?**

Enhancement in the Error Handling mechanism

**Did you add tests for your changes?**
No tests were required for the changes that are introduced.
**If relevant, did you update the documentation?**
there is a need to update the documentation

The pull request will resolve the issue #178 .

### CODE REVIEW

**Comment by iamsaumya** [comment]
Hey @somya339 , Do you mind if I review your PR? I was about the get this resolved, now as you are working on this issue I will like to extend my helping hand.

**Comment by Sagar2366** [comment]
@somya339 good job. Can you please add issue number with this PR.

**Comment by somya339** [comment]
@Sagar2366 now there is a need to update the documentation to follow the error throwing mechanism , should I create another pull request for that?

**Comment by Sagar2366** [comment]
> @Sagar2366 now there is a need to update the documentation to follow the error throwing mechanism , should I create another pull request for that?

Where do you want to update documentation ?

**Comment by somya339** [comment]
By Documentation I mean we have to Demonstrate  the use of this middleware somewhere such that it can be seen by all the contributors so that everyone is using " Apperror() " to throw error.

**Comment by palisadoes** [comment]
@somya339  Please resolve the merge conflicts

**Comment by palisadoes** [comment]
@somya339 We have since migrated to react. Is this PR in react?

File: error_middleware/error_handler.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = function AppError(message, httpStatus = 404) {
```
Comment (impiyush83) [inline error_middleware/error_handler.js:3]: AppError looks really meaningless and httpStatus = 404 assuming httpStatus as 404 is wrong. 

In error_middleware, 

Add various files  like,
internal-server-error.js - default HTTP  code - 500
not-found-error.js -  default HTTP  code - 404
conflict-error.js  - default HTTP  code - 409

**Review by impiyush83 — CHANGES_REQUESTED** [review]
Add error handlers which align to HTTP error codes

**Comment by palisadoes** [comment]
Closing this pull request due to a lack of response from the original author.

**Pull request closed without merging**